narrative_llm_agent/agents/analyst.py

```diff
@@ -6,62 +6,138 @@
+        description="query to look up KBase documentation, catalog or tutorials"
+    )
+
+DEFAULT_CATALOG_DB_DIR: Path = Path(__file__).parent / "vector_db_app_catalog"
+DEFAULT_DOCS_DB_DIR: Path = Path(__file__).parent / "vector_db_kbase_docs"
 
 class AnalystAgent(KBaseAgent):
-    role="Computational Biologist and Geneticist"
-    goal="Analyze and interpret datasets, and make suggestions into next analysis steps."
-    backstory="""You are an expert academic computational biologist with decades of
+    role = "Computational Biologist and Geneticist"
+    goal = (
+        "Analyze and interpret datasets, and make suggestions into next analysis steps."
+    )
+    backstory = """You are an expert academic computational biologist with decades of
     experience working in microbial genetics. You have published several genome announcement
     papers and have worked extensively with novel sequence data."""
+    _openai_key: str
+    _catalog_db_dir: Path
+    _docs_db_dir: Path
 
-    def __init__(self: "AnalystAgent", token: str, llm: LLM):
+    def __init__(
+        self: "AnalystAgent",
+        token: str,
+        llm: LLM,
+        openai_api_key: str = None,
+        catalog_db_dir: Path | str = None,
+        docs_db_dir: Path | str = None,
+    ):
         super().__init__(token, llm)
+        self.__setup_openai_api_key(openai_api_key)
+
+        if catalog_db_dir is not None:
+            self._catalog_db_dir = Path(catalog_db_dir)
+        else:
+            self._catalog_db_dir = DEFAULT_CATALOG_DB_DIR
+
+        if docs_db_dir is not None:
+            self._docs_db_dir = Path(docs_db_dir)
+        else:
+            self._docs_db_dir = DEFAULT_DOCS_DB_DIR
+
+        for db_path in [self._catalog_db_dir, self._docs_db_dir]:
+            self.__check_db_directories(db_path)
         self.__init_agent()
 
-    def __init_agent(self: "AnalystAgent"):
+    def __check_db_directories(self, db_path: Path) -> None:
+        """
+        Checks for presence of the expected database directory. Doesn't look for all files,
+        just ensures that the directory is present and it has a `chroma.sqlite3` file.
+        If file or directory is missing, this raises a RuntimeError.
+        TODO: check for other files if needed
+        """
+        if not db_path.exists():
+            raise RuntimeError(
+                f"Database directory {db_path} not found, unable to make Agent."
+            )
+        if not db_path.is_dir():
+            raise RuntimeError(
+                f"Database directory {db_path} is not a directory, unable to make Agent."
+            )
+        db_file = db_path / "chroma.sqlite3"
+        if not db_file.exists():
+            raise RuntimeError(
+                f"Database file {db_file} not found, unable to make Agent."
+            )
 
-        @tool("Kbase documentation retrieval tool", args_schema = AnalystInput, return_direct=True)
+    def __setup_openai_api_key(self, openai_api_key: str) -> None:
```

Comment:
If we have a key as a parameter, great. If not, look for the environment variable. If that doesn't exist either, throw an error.